### ISSUE TITLE
GetSource returns invalid text after navigation to Cyrillic sites

### ISSUE THREAD

**Issue by AStrelkov** [MicrosoftEdge/WebView2Feedback#1427]
**Description**
Right away after invocation of method **Navigate(L"https:\\честныйзнак.рф")**
following methods return text _https://xn--80ajghhoc2aj1c8b.xn--p1ai/_ :
- webView->GetSource() 
- ICoreWebView2NavigationStartingEventArgs -> get_Uri()

Sometimes the result of navigation with  _url_ name, that has Cyrillic characters in its path, is GPF error. With english site/url names the method works fine.

**Version**
SDK: 1.0.864.35
Runtime: EverGreen 91.0.864.48 -->
Framework: C++ 2017, Express
OS: Win7 64 bit Pro

**Comment by champnic** [comment]
Hey @AStrelkov - the "xn--..." is the punycode of the url: https://en.wikipedia.org/wiki/Punycode

If you'd like to convert back to unicode you could use the [IdnMapping ](https://docs.microsoft.com/en-us/dotnet/api/system.globalization.idnmapping?redirectedfrom=MSDN&view=net-5.0)class in C# - I'm not sure what the equivalent is in C++.

What are the errors you are getting with certain urls? Do you have examples of that?

**Comment by AStrelkov** [comment]
Well, but it would be very nice that WebView2 itself makes such decoding from PunyCode. To my mind embedding this functionality into component is more right way.

**Comment by champnic** [comment]
@AStrelkov Agreed it would be nice to keep the punycode as an internal implementation detail - I've added this as a feature request on our backlog. Thanks!

**Comment by AStrelkov** [comment]
**@champnic** OK, thanks, I will wait for the update!
